### PR TITLE
implementation of identity service verification procedure

### PR DESCRIPTION
**Pull request by Ishikabaid**
Before clicking on the generate chaincode button
Screenshot (114)

After generating the chaincode
Screenshot (115)

### CODE REVIEW

File: app/controllers/identity.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  @action async handleGenerateChaincode(e) {
+    e.preventDefault();
+    this.isGenerateChaincodeClicked = true;
+
+    try {
+      const response = await fetch(`${BASE_URL}/users/chainCode`, {
+        method: 'PATCH',
+        body: JSON.stringify({}),
```
Comment (lakshayman) [inline app/controllers/identity.js:75]: We can remove this body line.

File: app/controllers/identity.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  @action handleVerify() {
+    if (this.isChecked === false) {
+      alert('Please verify!');
+    } else {
+      alert('Approved!');
+    }
+  }
+
+  @action handleChaincode() {
```
Comment (lakshayman) [inline app/controllers/identity.js:36]: Tomorrow the APIs hopefully get merged so we can merge this function and below generate chain code function

**Review by lakshayman — CHANGES_REQUESTED** [review]

File: app/controllers/identity.js (hunk 1 above)
Comment (Ishikabaid) [reply app/controllers/identity.js:75]: Okay!

File: app/controllers/identity.js (hunk 2 above)
Comment (Ishikabaid) [reply app/controllers/identity.js:36]: Yeah sure.

File: app/controllers/identity.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      });
+
+      const { status } = response;
+      if (status === 204) {
+        alert('Identity URL successfully edited');
+      } else {
+        alert('Something went wrong. Please check console errors.');
+      }
+    } catch (error) {
+      console.error('Error : ', error);
```
Comment (swarajpure) [inline app/controllers/identity.js:62]: Let's add an alert here, something like "Oops, something went wrong"

File: package-lock.json [hunk 4]
```diff
@@ -5,8 +5,7 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "website-my",
-      "version": "1.1.0",
+      "version": "0.0.0",
```
Comment (lakshayman) [inline package-lock.json:8]: package-lock.json should not update. I think
cc: @swarajpure

File: app/controllers/identity.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+const BASE_URL = ENV.BASE_API_URL;
+
+export default class IdentityController extends Controller {
+  @tracked isEditClicked = false;
+  @tracked isGenerateChaincodeClicked = false;
+  @tracked identityURL = '';
+  @tracked isChecked = false;
+  @tracked isChaincodeClicked = false;
+  @tracked Chaincode = 'Generate Chaincode';
+  @tracked Code = 'Copy';
```
Comment (lakshayman) [inline app/controllers/identity.js:15]: Unused variable

File: app/templates/identity.hbs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      {{else}}
+       <div class="input-div">
+       <label>Identity Url</label>
+       <Input @value={{@model.identityURL}} type="text" placeholder="service url" />
+       <button    
+        type="submit"
+        class="identity-Buttons edit"
+        {{on 'click' (fn this.handleEdit)}}>
+        Edit
+        </button>
```
Comment (lakshayman) [inline app/templates/identity.hbs:20]: Need to fix some spacing here

**Review by lakshayman — CHANGES_REQUESTED** [review]

File: app/controllers/identity.js (hunk 3 above)
Comment (lakshayman) [reply app/controllers/identity.js:58]: @Ishikabaid

**Comment by Ishikabaid** [comment]
Continued in this [PR](https://github.com/Real-Dev-Squad/website-my/pull/224)

**Pull request closed without merging**